<?xml version="1.0" encoding="utf-8"?>

<launch>
  <param name="/use_sim_time" value="true"/>
  <rosparam file="$(find lidar_odo)/config/params.yaml" command="load" />
  <node pkg="lidar_odo" type="lidar_odo_imuPreintegration"   name="lidar_odo_imuPreintegration"    output="screen" 	respawn="true"/>
  <node pkg="lidar_odo" type="lidar_odo_imageProjection"     name="lidar_odo_imageProjection"      output="screen"     respawn="true"/>
  <node pkg="lidar_odo" type="lidar_odo_mapOptmization"      name="lidar_odo_mapOptmization"       output="screen"     respawn="true"/>

  <arg name="project" default="cvr_lse"/>
  <node name="cvr_lse" pkg="cvr_lse" type="cvr_lse" output="screen">
    <rosparam command="load" file="$(find cvr_lse)/config/cvr_lse_params.yaml"/>
    <param name = "lidar_topic" value = "/hesai40p_points_xyzirt" />
    <param name = "cloud_label_topic" value = "/ground_segmentation/cloud_label"/>
    <param name = "lidar_odom_topic" value = "/odometry/imu" />
    <param name = "ground_output_topic" value = "/cvr_lse/ground_cloud" />
    <param name = "obstacle_output_topic" value = "/cvr_lse/obstacle_cloud" />
  </node>

  <node name="ground_segmentation" pkg="cvr_lse" type="ground_segmentation" output="screen">
    <rosparam command="load" file="$(find cvr_lse)/config/segmentation_params.yaml"/>
    <param name = "lidar_topic" value = "/hesai40p_points_xyzirt" />
    <param name = "cloud_label_topic" value = "/ground_segmentation/cloud_label" />
  </node>
  <node pkg="rviz" type="rviz" name="$(arg project)_rviz" args="-d $(find cvr_lse)/config/rviz.rviz" />
</launch>
